src/core/createPayload.js

```diff
@@ -10,60 +10,28 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-/* TODO: Remove.
+import createPayloadItemMerger from "./createPayloadItemMerger";
 
-  Current Request Schema:
-
-  https://git.corp.adobe.com/experience-edge/ue-gateway/
-  blob/b946662d672950898248daf346ff6adea4d41de4/resources/request.json
-
-  Top-Level nodes:
-
-  {
-    query: {},
-    "context": {
-      "identityMap": {},
-      "environment": {},
-      "webreferrer": {}
-    },
-    "events": [{ // Might contain meta per event.
-      "timestamp": 1550574782,
-      "eventId": "test",
-      "correlationID": "something",
-      "type": "::page:load",
-      "data": {}
-    }],
-    "meta": {}
-  }
-
-*/
-
-import { assign } from "../utils";
-
-const append = (content, key) => (obj = {}) => {
-  // TODO Validate.
-  console.warn(`[Payload:appendTo${key}] To Implement!`);
-  assign(content[key], obj);
-  return content;
-};
-
-export default ({
-  events = [],
-  query = {},
-  metadata = {},
-  context = {}
-} = {}) => {
-  const content = { events, query, metadata, context };
+export default () => {
+  const content = {};
 
   return {
-    addEvent(ev) {
-      content.events.push(ev);
+    setPropertyID(propertyID) {
+      content.propertyID = propertyID;
+    },
+    addIdentity: (namespaceCode, identity) => {
+      content.identityMap = content.identityMap || {};
+      content.identityMap[namespaceCode] =
+        content.identityMap[namespaceCode] || [];
+      content.identityMap[namespaceCode].push(identity);
+    },
+    addEvent(event) {
+      content.events = content.events || [];
+      content.events.push(event);
     },
-    addQuery: append(content, "query"),
-    addMetadata: append(content, "metadata"),
-    addContext: append(content, "context"),
-    toJson() {
-      return JSON.stringify(content);
+    mergeMeta: createPayloadItemMerger(content, "meta"),
+    toJSON() {
+      return content;
```

Comment:
`JSON.stringify()`, which is later used to serialize the payload, looks for [a `toJSON` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON()_behavior) in order to determine how to serialize an object. Were weren't taking advantage of this previously (we were manually calling `toJson()`).